packages/react-core/src/components/Button/Button.tsx

```diff
@@ -95,6 +95,8 @@ export interface ButtonProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'r
   innerRef?: React.Ref<any>;
   /** Adds count number to button */
   countOptions?: BadgeCountObject;
+  /** @hide Sets the role of the button. Should only be used when the button is a descendant of a menu or tablist. */
+  role?: string;
```

Comment:
I'm hesitant to expose this publicly since it's a very specific use-case. I originally had just imported the button styles into MenuItemAction and applied the necessary class names to a native <button> element, so we could do that instead.

This could be worth exposing to the consumer once we can create some demos on when/how to properly use the prop.

Reply:
I think it's probably fine to remove this and leave it as an internal prop until we need it to be exposed. But hiding it also works for me.

Reply:
True, we could move the spread props in button to come after all the other props (currently they're spread first). Wasn't sure if there was a particular reason for that or if it's safe to just move that.

https://github.com/patternfly/patternfly-react/blob/08a6b1aaaf6769b657ed674f4bb741b530265c7d/packages/react-core/src/components/Button/Button.tsx#L146-L152

Reply:
@kmcfaul updated